src/wallet/wallet.cpp

```diff
@@ -3016,9 +3057,27 @@ int64_t CWallet::GetOldestKeyPoolTime()
     if (setKeyPool.empty())
         return GetTime();
 
-    // load oldest key from keypool, get time and return
     CKeyPool keypool;
     CWalletDB walletdb(strWalletFile);
+
+    if (IsHDEnabled() && CanSupportFeature(FEATURE_HD_SPLIT))
+    {
+        // if HD & HD Chain Split is enabled, response max(oldest-internal-key, oldest-external-key)
+        int64_t now = GetTime();
+        int64_t oldest_external = now, oldest_internal = now;
+
+        for(const int64_t& id : setKeyPool)
+        {
+            if (!walletdb.ReadPool(id, keypool))
+                throw std::runtime_error(std::string(__func__) + ": read failed");
+            if (keypool.fInternal && keypool.nTime < oldest_internal)
+                oldest_internal = keypool.nTime;
+            else if (!keypool.fInternal && keypool.nTime < oldest_external)
+                oldest_external = keypool.nTime;
```

Comment:
Ah. Right. It's guaranteed that the `setKeyPool` iterates always from the oldest to the newest.
Will add.